lib/spack/spack/compilers/rocmcc.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import re
+
+import llnl.util.lang
+
+import spack.compilers.clang
+
+
+class Rocmcc(spack.compilers.clang.Clang):
+    # Subclasses use possible names of C compiler
+    cc_names = ['amdclang']
+
+    # Subclasses use possible names of C++ compiler
+    cxx_names = ['amdclang++']
+
+    # Subclasses use possible names of Fortran 77 compiler
+    f77_names = ['amdflang']
+
+    # Subclasses use possible names of Fortran 90 compiler
+    fc_names = ['amdflang']
+
+    PrgEnv = 'PrgEnv-amd'
+    PrgEnv_compiler = 'amd'
+
+    @property
+    def link_paths(self):
+        link_paths = {'cc': 'rocmcc/clang',
+                      'cxx': 'rocmcc/clang++',
+                      'f77': 'rocmcc/flang',
+                      'fc': 'rocmcc/flang'}
```

Comment:
I think you want to prepend the wrappers with `amd`
e.g. 
```
        link_paths = {'cc': 'rocmcc/amdclang',
                      'cxx': 'rocmcc/amdclang++',
                      'f77': 'rocmcc/amdflang',
                      'fc': 'rocmcc/amdflang'}
```

Reply:
@becker33  one more issue I encountered during testing concerns `lib/spack/env/cc`. I fixed the issue in #28579